### PR TITLE
Add an option to enable ubsan

### PR DESCRIPTION
**Pull request by tobim**
###  :notebook_with_decorative_cover: Description

See title. Be aware that compilation times go up by a lot, that's why I did not add it to `dev-mode` or `ci-build`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -322,6 +322,8 @@ else ()
   message(FATAL_ERROR "Unsupported compiler: ${CMAKE_CXX_COMPILER_ID}")
 endif ()
 
+option(ENABLE_ADDRESS_SANITIZER
+       "Insert pointer and reference checks into the generated binaries" OFF)
```
Comment (mavam) [inline CMakeLists.txt:326]: Just curious? What does this unrelated change do?

Comment (dominiklohmann) [reply CMakeLists.txt:326]: It just exposes an existing option.

**Review by dominiklohmann — APPROVED** [review]
This looks great, thanks for this. I ran this locally and it works as expected with clang-11.
